packages/graalvm/src/main/kotlin/elide/runtime/feature/engine/AbstractStaticNativeLibraryFeature.kt

```diff
@@ -171,7 +171,9 @@ public abstract class AbstractStaticNativeLibraryFeature : NativeLibraryFeature
       if (it.registerJni) (access as BeforeAnalysisAccessImpl).nativeLibraries.let { nativeLibraries ->
         when (it.type) {
           STATIC -> nativeLibraries.addStaticJniLibrary(it.name)
-          SHARED -> error("Dynamic native libraries not supported yet: $it")
+          SHARED -> if (it.registerJni) { /* Dynamic JNI libraries use JNI to load. */ } else {
+            nativeLibraries.addDynamicNonJniLibrary(it.name)
+          }
```

Comment:
Is there a point to having the empty `if` branch? why not invert the condition and keep the comment?

Reply:
@darvld good point -- should get compiled out since `it.registerJni` is known at compile time (this is all build time code anyway) but will catch in the next PR. there is going to be logic in that other branch eventually, i think

Reply:
I agree that the compiler will optimize it out, I was just checking if you had forgotten to add some logic there 👍